Link Implants to URLs and show URLID in handler

riskydissonance opened the pull request:
* URL info can be viewed with show-urls
* Added classes to Models.py and returning those from DB to more easily handle Implants/C2 etc.
* Some variable renaming for clarity

* [x]  TODO enforce unique URL names
* [x] TODO test on Postgres

riskydissonance committed:
Add generate-csvs command to generate CSV reports only.

removed typo


Don't need port for debug command

Add Postgres csv output

riskydissonance committed:
Cleanup Sharp PowerShell Runner

PwnDexter commented on poshc2/client/command_handlers/ImplantHandler.py:
```diff
@@ -206,7 +207,7 @@ def implant_handler_command_loop(user, printhelp="", autohide=None):
                 continue
             if command.startswith("createproxypayload"):
-                do_createproxypayload(user, command)
+                do_createnewpayload(user, command)
```
Does this/How will this affect the invoke-XXXproxypayload commands etc? i.e., invoke-wmiproxypayload

benpturner committed:
Updated kill date

riskydissonance commented:
TODO
* [x] Remove PayloadCommsPort
* [x] Check invoke-*proxy payloads

benpturner commented:
- [x] Add the Payload offset to this branch

benpturner committed:
Link Implants to URLs and show URLID in handler

URL info can be viewed with show-urls
Added classes to Models.py and returning those from DB to more easily handle Impants/C2 etc.

Enforce unique URL names

Fixed error in C2 Model via SQLite


Generate URLs report


Fixed createdaisypayload


Updated kill-implant


Fixed start of postgres stuff


Fixed DB Error Postgres


Fixed reporting bug


Fixed daisy payload generation


Fixed Python dropper


Fixed bug in QuickCommand


Link Implants to URLs and show URLID in handler

URL info can be viewed with show-urls
Added classes to Models.py and returning those from DB to more easily handle Impants/C2 etc.

Removed PayloadCommsPort


dsa

benpturner committed:
Added offset finder

benpturner committed:
Consolidate & Clean Up invoke-*payload commands

benpturner committed:
Merged get-system and install-persistence
